### ISSUE TITLE
Collection Pages : Pagination lacking descriptive link text

### ISSUE THREAD

**Issue by mbambara** [Shopify/dawn#683]
**Current behavior:**
The page navigation numbers are created in span tags that use the aria-label attribute to describe the page number. The aria-label attribute cannot be used in span tags and defaults to reading the content in the span tags, deeming the aria-label ineffective.

**Possible solution:**
Remove aria-label attribute with the potential to replace with aria-hidden. 

**Version information:**
- Dawn Version: 2.1.0
- Chrome Version 93.0.4577.8
- macOS Version 10.14.6
